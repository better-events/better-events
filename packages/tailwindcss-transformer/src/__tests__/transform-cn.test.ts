import { describe, expect, it } from "vitest";
import { transform } from "../index";

describe("cn/cx/clsx", () => {
	it("should transform correctly", () => {
		const result = transform(
			`const ButtonCn: React.FC<ButtonProps> = ({ children, className }) => {
        const buttonClasses = cn(
          (something === 'flex' || 'flex') && 'flex',
          ['flex', 'flex'],
          something === 'flex' || something === 'flex' ? (something ? 'flex' : 'flex') : 'flex',
          className,
        );
        return <button className={buttonClasses}>{children}</button>;
      };
      
      const ButtonCx: React.FC<ButtonProps> = ({ children, className }) => {
        const buttonClasses = cx(
          (something === 'flex' || 'flex') && 'flex',
          ['flex', 'flex'],
          something === 'flex' || something === 'flex' ? (something ? 'flex' : 'flex') : 'flex',
          className,
        );
        return <button className={buttonClasses}>{children}</button>;
      };
      
      const ButtonClsx: React.FC<ButtonProps> = ({ children, className }) => {
        const buttonClasses = clsx(
          (something === 'flex' || 'flex') && 'flex',
          ['flex', 'flex'],
          something === 'flex' || something === 'flex' ? (something ? 'flex' : 'flex') : 'flex',
          className,
        );
        return <button className={buttonClasses}>{children}</button>;
      };`,
			{
				styleCache: new Map(),
			},
		);
		expect(result).toMatchSnapshot();
	});

	it("should test cnExt logic correctly", () => {
		const result = transform(
			`const ButtonCnExt: React.FC<ButtonProps> = ({ children, className }) => {
        const buttonClasses = cnExt(
          (something === 'flex' || 'flex') && 'flex',
          ['flex', 'flex'],
          something === 'flex' || something === 'flex' ? (something ? 'flex' : 'flex') : 'flex',
          className,
        );
        return <button className={buttonClasses}>{children}</button>;
      };`,
			{
				styleCache: new Map(),
			},
		);
		expect(result).toMatchSnapshot();
	});
});
