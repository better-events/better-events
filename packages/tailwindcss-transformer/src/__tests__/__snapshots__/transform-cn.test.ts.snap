// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`cn/cx/clsx > should test cnExt logic correctly 1`] = `
"const ButtonCnExt: React.FC<ButtonProps> = ({ children, className }) => {
        const buttonClasses = cnExt(
          (something === 'flex' || 'flex') && "kf-222f930b",
          ["kf-222f930b", "kf-222f930b"],
          something === 'flex' || something === 'flex' ? (something ? "kf-222f930b" : "kf-222f930b") : "kf-222f930b",
          className,
        );
        return <button className={buttonClasses}>{children}</button>;
      };"
`;

exports[`cn/cx/clsx > should transform correctly 1`] = `
"const ButtonCn: React.FC<ButtonProps> = ({ children, className }) => {
        const buttonClasses = cn(
          (something === 'flex' || 'flex') && "kf-222f930b",
          ["kf-222f930b", "kf-222f930b"],
          something === 'flex' || something === 'flex' ? (something ? "kf-222f930b" : "kf-222f930b") : "kf-222f930b",
          className,
        );
        return <button className={buttonClasses}>{children}</button>;
      };
      
      const ButtonCx: React.FC<ButtonProps> = ({ children, className }) => {
        const buttonClasses = cx(
          (something === 'flex' || 'flex') && "kf-222f930b",
          ["kf-222f930b", "kf-222f930b"],
          something === 'flex' || something === 'flex' ? (something ? "kf-222f930b" : "kf-222f930b") : "kf-222f930b",
          className,
        );
        return <button className={buttonClasses}>{children}</button>;
      };
      
      const ButtonClsx: React.FC<ButtonProps> = ({ children, className }) => {
        const buttonClasses = clsx(
          (something === 'flex' || 'flex') && "kf-222f930b",
          ["kf-222f930b", "kf-222f930b"],
          something === 'flex' || something === 'flex' ? (something ? "kf-222f930b" : "kf-222f930b") : "kf-222f930b",
          className,
        );
        return <button className={buttonClasses}>{children}</button>;
      };"
`;
